### ISSUE TITLE
Admin Switch-User Functionality

### ISSUE THREAD

**Issue by danrichards** [pyrocms/pyrocms#2922]
Another simple feature request...maybe

Use Case 1, current: 
-Admin user is logged out.
-Direct URL to module/feature is accessed.
-User logs in and is automatically to desired module/feature 

Use Case 2, desired:
-Organization has modules with different permission levels
-L1 Admin User is already logged in.
-Module feature requires L2 Admin User, Switch User anchor is provided
-Proposed Anchor example: admin/logout/orders/rhodyready/open/4516
-System logs user out and redirects to login
-L2 Admin User logs in and is redirected to desired module/function/whatever

Use Case 3, oh so awesome
-Settings in Users core has option to enable "Switch User" menu.
-Profile menu link has "Switch User" option.
-Link is dynamic to whatever module/feature you're on in the admin section.
-So in other words, you could be in admin/pages/edit/14, goto menu > Profile > Switch User, then be logged out, log-in, and redirected back to the same place.

Use Case 4, let's go banana sandwich
"Oh so awesome" functionality uses Colorbox modal to sign-in to another account. Page below model refreshes after sign in.
